torchvision/ops/misc.py

```diff
@@ -70,20 +70,23 @@ def __init__(
         self,
         in_channels: int,
         out_channels: int,
-        kernel_size: int = 3,
-        stride: int = 1,
-        padding: Optional[int] = None,
+        kernel_size: Union[int, Tuple[int, ...]] = 3,
+        stride: Union[int, Tuple[int, ...]] = 1,
+        padding: Optional[Union[int, Tuple[int, ...]]] = None,
         groups: int = 1,
         norm_layer: Optional[Callable[..., torch.nn.Module]] = torch.nn.BatchNorm2d,
         activation_layer: Optional[Callable[..., torch.nn.Module]] = torch.nn.ReLU,
-        dilation: int = 1,
+        dilation: Union[int, Tuple[int, ...]] = 1,
         inplace: Optional[bool] = True,
         bias: Optional[bool] = None,
         conv_layer: Callable[..., torch.nn.Module] = torch.nn.Conv2d,
     ) -> None:
 
+        _conv_dim = 2 if conv_layer == torch.nn.Conv2d else 3
```

Comment:
This is a bit flimsy way to capture the dimensionality. What if we have a different type of 2d [convolution](https://pytorch.org/docs/stable/nn.html#convolution-layers) like `LazyConv2d` or `ConvTranspose2d`?

I think one way to avoid this is by not requiring to find this number. We can rely on the user to provide us with either single integer and keep it that way or a tuple.

Reply:
I have updated the code, to get convolution dimension from `kernel_size` or `dilation` if needed. What do you think on this?